clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -472,14 +475,63 @@ static Value *EmitISOVolatileStore(CodeGenFunction &CGF, const CallExpr *E) {
   return Store;
 }
 
+static StringRef
+convertFPAccuracy(LangOptions::FPAccuracyKind FPAccuracy) {
+  StringRef AccuracyVal;
+  switch (FPAccuracy) {
+  case LangOptions::FPA_Default:
+    AccuracyVal = "fpaccuracy.default";
+    break;
+  case LangOptions::FPA_High:
+    AccuracyVal = "fpaccuracy.high";
+    break;
+  case LangOptions::FPA_Medium:
+    AccuracyVal = "fpaccuracy.medium";
+    break;
+  case LangOptions::FPA_Low:
+    AccuracyVal = "fpaccuracy.low";
+    break;
+  case LangOptions::FPA_Sycl:
+    AccuracyVal = "fpaccuracy.sycl";
+    break;
+  case LangOptions::FPA_Cuda:
+    AccuracyVal = "fpaccuracy.cuda";
+    break;
+  }
+  return AccuracyVal;
+}
+
 // Emit a simple mangled intrinsic that has 1 argument and a return type
 // matching the argument type. Depending on mode, this may be a constrained
 // floating-point intrinsic.
 static Value *emitUnaryMaybeConstrainedFPBuiltin(CodeGenFunction &CGF,
                                 const CallExpr *E, unsigned IntrinsicID,
-                                unsigned ConstrainedIntrinsicID) {
+                                unsigned ConstrainedIntrinsicID,
+    unsigned FPAccuracyIntrinsicID = Intrinsic::not_intrinsic) {
+  llvm::TargetLibraryInfoImpl TLII(CGF.getTarget().getTriple());
```

Comment:
I don't think this will work. We'll need to populate the alternate math library info in the createTLII() function in BackendUtil.cpp. What you'll get this way will be a different TLII instance that doesn't have the alternate math library info populated.

Reply:
mh!  I don't see how FE can use TLII after the BackendUtil.cpp has populated with alternate functions? We need another mechanism.